DataFormats/HcalRecHit/interface/HFPreRecHit.h

```diff
@@ -0,0 +1,38 @@
+#ifndef DATAFORMATS_HCALRECHIT_HFPRERECHIT_H
+#define DATAFORMATS_HCALRECHIT_HFPRERECHIT_H
+
+#include "DataFormats/HcalRecHit/interface/HFQIE10Info.h"
+
+/** \class HFPreRecHit
+*
+* Class to contain the info needed to perform HF reconstruction
+* using QIE10 chips and dual-anode readout. Groups the information
+* provided by a single PMT.
+*/
+class HFPreRecHit
+{
+public:
+    typedef HcalDetId key_type;
+
+    HFPreRecHit();
+
+    // In the constructor below, either "first" or "second"
+    // pointer can be nullptr
+    HFPreRecHit(const HcalDetId& id, const HFQIE10Info* first,
+                const HFQIE10Info* second);
+
+    inline HcalDetId id() const {return id_;}
+
+    // Get a pointer to the QIE10 info. nullptr will be returned
+    // if the info with the given index does not exist or if the
+    // index is out of range.
+    const HFQIE10Info* getHFQIE10Info(unsigned index) const;
+
+private:
+    HcalDetId id_;
+
+    HFQIE10Info hfQIE10Info_[2];
+    uint8_t hasInfo_[2];
```

Comment:
Please make sure to mention next time everything you
are waiting for. Enforcing your personal preferences
on somebody else's perfectly functional code in a very
obscure manner slows down the code development
process significantly.

On 05/09/2016 03:11 PM, Slava Krutelyov wrote:
>
> In DataFormats/HcalRecHit/interface/HFPreRecHit.h <https://github.com/cms-sw/cmssw/pull/14383#discussion_r62564272>:
>
> > +    // pointer can be nullptr
> > +    HFPreRecHit(const HcalDetId& id, const HFQIE10Info* first,
> > +                const HFQIE10Info* second);
> > +
> > +    inline HcalDetId id() const {return id_;}
> > +
> > +    // Get a pointer to the QIE10 info. nullptr will be returned
> > +    // if the info with the given index does not exist or if the
> > +    // index is out of range.
> > +    const HFQIE10Info* getHFQIE10Info(unsigned index) const;
> > +
> > +private:
> > +    HcalDetId id_;
> > +
> > +    HFQIE10Info hfQIE10Info_[2];
> > +    uint8_t hasInfo_[2];
>
> I was actually waiting for a change to |bool|
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/cms-sw/cmssw/pull/14383/files/d13235f9c646cc3c33425064ddf2551a81a4bef7#r62564272>
>